source/Nevermore/Advanced/WriteTransaction.cs

```diff
@@ -238,59 +252,118 @@ public async Task CommitAsync(CancellationToken cancellationToken = default)
             await configuration.Hooks.AfterCommitAsync(this);
         }
 
-        object[] ExecuteDataModification(PreparedCommand command)
+        DataModificationOutput[] ExecuteDataModification(PreparedCommand command)
         {
-            if (!command.Mapping.IsRowVersioningEnabled)
+            if (!command.Mapping.HasModificationOutputs)
```

Comment:
there are 2 different possible modification outputs, the RowVersion and/or the new Id